build/helpers/file-includes-all-subdirectories-as-named-exports.ts

```diff
@@ -0,0 +1,33 @@
+import * as path from "path";
+import * as fs from "fs";
+import { pascalCase } from "../../packages/fast-web-utilities";
+
+/**
+ * Verifies all components in source directory are exported
+ */
+export function includesAllSubdirectoriesAsNamedExports(indexFile: string): boolean | Error {
+    /**
+     * Get the folders in the indexFile directory to compare to export listing
+     */
+    const directoryPath: string = path.dirname(indexFile);
+    const components: string[] = [];
+    const sourceDirectories: string[] = fs.readdirSync(directoryPath).map(
+        (name: string) => path.join(directoryPath, name)).filter(
+            (source: string) => fs.statSync(source).isDirectory());
+
+    sourceDirectories.forEach((entry: string) => {
```

Comment:
You can avoid creating both a `sourceDirectories` and `components` variables:

```
const components = fs.readDirSync(directoryPath)
    .map(/*...*/)
    .filter(/*...*/)
    .map((entry: string) => PascalCase(path.basename(entry)))`
```

Reply:
Sure thing